lib/LWP/Protocol/https.pm

```diff
@@ -55,79 +55,12 @@ EOT
     return (%ssl_opts, $self->SUPER::_extra_sock_opts);
 }
 
-#------------------------------------------------------------
-# _cn_match($common_name, $san_name)
-#  common_name: an IA5String
-#  san_name: subjectAltName
-# initially we were only concerned with the dNSName
-# and the 'left-most' only wildcard as noted in
-#   https://tools.ietf.org/html/rfc6125#section-6.4.3
-# this method does not match any wildcarding in the
-# domain name as listed in section-6.4.3.3
-#
-sub _cn_match {
-    my( $me, $common_name, $san_name ) = @_;
-
-    # /CN has a '*.' prefix
-    # MUST be an FQDN -- fishing?
-    return 0 if( $common_name =~ /^\*\./ );
-
-    my $re = q{}; # empty string
-
-     # turn a leading "*." into a regex
-    if( $san_name =~ /^\*\./ ) {
-        $san_name =~ s/\*//;
-        $re = "[^.]+";
-    }
-
-      # quotemeta the rest and match anchored
-    if( $common_name =~ /^$re\Q$san_name\E$/ ) {
-        return 1;
-    }
-    return 0;
-}
-
-#-------------------------------------------------------
-# _in_san( cn, cert )
-#  'cn' of the form  /CN=host_to_check ( "Common Name" form )
-#  'cert' any object that implements a peer_certificate('subjectAltNames') method
-#   which will return an array of  ( type-id, value ) pairings per
-#   http://tools.ietf.org/html/rfc5280#section-4.2.1.6
-# if there is no subjectAltNames there is nothing more to do.
-# currently we have a _cn_match() that will allow for simple compare.
-sub _in_san
-{
-    my($me, $cn, $cert) = @_;
-
-	  # we can return early if there are no SAN options.
-	my @sans = $cert->peer_certificate('subjectAltNames');
-	return unless scalar @sans;
-
-	(my $common_name = $cn) =~ s/.*=//; # strip off the prefix.
-
-      # get the ( type-id, value ) pairwise
-      # currently only the basic CN to san_name check
-    while( my ( $type_id, $value ) = splice( @sans, 0, 2 ) ) {
-        return 'ok' if $me->_cn_match($common_name,$value);
-    }
-    return;
-}
-
+# This is a subclass of LWP::Protocol::http.
+# That parent class calls ->_check_sock() during the
```

Comment:
This doesn't appear to be true anymore?

I think it's safe to remove _check_sock entirely, since it would only be called via OO and there is a stub in the parent class just in case anything is calling it, although nothing should be.

Reply:
Right. I left the stub in this class solely to have a place to write that tidbit of docs. Should we remove that all together and just document it with the commit message?

Reply:
I think it'll be easier to find that documentation in the code rather than in a commit message.